src/udev/net/ethtool-util.c

```diff
@@ -206,3 +212,108 @@ int ethtool_set_wol(int *fd, const char *ifname, WakeOnLan wol) {
 
         return 0;
 }
+
+static int ethtool_get_stringset(int *fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **gstrings) {
+        _cleanup_free_ struct ethtool_gstrings *strings = NULL;
+        struct ethtool_sset_info info =  {
+                .cmd = ETHTOOL_GSSET_INFO,
+                .reserved = 0,
+                .sset_mask = 1ULL << stringset_id,
+        };
+        unsigned len;
+        int r;
+
+        ifr->ifr_data = (void *) &info;
+
+        r = ioctl(*fd, SIOCETHTOOL, ifr);
+        if (r < 0)
+                return -errno;
+
+        if (!info.sset_mask)
+                return -EINVAL;
+
+        len = info.data[0];
+
+        strings = malloc0(sizeof(struct ethtool_gstrings) + len * ETH_GSTRING_LEN);
+        if (!strings)
+                return log_oom();
```

Comment:
this function logs about OOM, but not about other errors. In systemd, we try to make that a function either logs about all errors or about none (in wich case the caller should log). Please make your mind up if  ethtool_get_stringset() should be of the logging kind or not. If it should log, then it needs to log in all cases. If it should not log, then in none but the caller should.